### PR TITLE
feat: add 404 not found page

### LINKED ISSUE #170: Add 404 page
### Clear and concise description of the problem

If going to any unexpected url e.g. `https://code-generator.pytorch-ignite.ai/test`, we see a blank page. 
Let's show a 404 page ?

### PR DESCRIPTION
**Pull request by afzal442**
Signed-off-by: afzal442 <afzal442@gmail.com>

### Description

Fix #170 

### Additional context

---

### What is the purpose of this pull request?

- [ ] Bug fix
- [x] New feature
- [ ] Other

### CODE REVIEW

**Comment by afzal442** [comment]
Plz, have a look at the screenshot;

c3

**Comment by trsvchn** [comment]
Thanks @afzal442!
Let me run the deploy preview!

**Comment by trsvchn** [comment]
@afzal442 Looks like we have some linting issues here

https://github.com/pytorch-ignite/code-generator/pull/178/checks?check_run_id=3781076007#step:9:13

Please take a look at our [CONTRIBUTING](https://github.com/pytorch-ignite/code-generator/blob/main/CONTRIBUTING.md#pull-request-guidelines) how to run formatters.

**Comment by trsvchn** [comment]
Thanks @afzal442 ! It works!
https://deploy-preview-178--code-generator.netlify.app/foo

But let's try to add header and a footer, to get smth like this:
404page

Inspired by [pytorch 404 page](https://pytorch.org/foo.bar)

**Comment by afzal442** [comment]
@trsvchn I am done. 🪧 Plz also add ` hacktoberfest` label as well. 😵‍💫

**Comment by trsvchn** [comment]
Thanks @afzal442 ! This looks much better :)

Let's split the page content into several lines, to fit into the following template:

```
Error Code

Error Short Description

Error Long Description

How to solve
```

<div align="center">

---

navbar

---

**404**

**Page Not Found**

Oops we couldn't find that page.

Try going [home]()

---

footer

---

</div>

File: src/views/PageNotFound.vue [hunk 1]
```diff
@@ -0,0 +1,62 @@
+<template>
+  <main>
+    <NavBar />
+    <div class="heading">
+      <h1>404</h1>
+      <h3>Page Not Found</h3>
+      <h3>Oops we couldn't find that page.</h3>
+      <p>
+        Try going
+        <RouterLink to="/" class="link alt" :style="learnMore">home</RouterLink>
```
Comment (trsvchn) [inline src/views/PageNotFound.vue:10]:
```suggestion
        <RouterLink to="/" class="link alt" :style="learnMore">homepage</RouterLink>
```

File: src/views/PageNotFound.vue [hunk 2]
```diff
@@ -0,0 +1,62 @@
+<template>
+  <main>
+    <NavBar />
+    <div class="heading">
+      <h1>404</h1>
+      <h3>Page Not Found</h3>
+      <h3>Oops we couldn't find that page.</h3>
+      <p>
+        Try going
```
Comment (trsvchn) [inline src/views/PageNotFound.vue:9]:
```suggestion
        Go back to the
```

**Review by trsvchn — COMMENTED** [review]
Thanks for the updates @afzal442 !

I left some nit comments related to wording.

File: src/views/PageNotFound.vue [hunk 3]
```diff
@@ -0,0 +1,64 @@
+<template>
+  <main>
+    <NavBar />
+    <div class="heading">
+      <h1>404</h1>
+      <h3>Page Not Found</h3>
+      <h3>Oops! we couldn't find that page.</h3>
```
Comment (trsvchn) [inline src/views/PageNotFound.vue:7]:
```suggestion
      <h3>Oops! We couldn't find that page.</h3>
```

File: src/main.js [hunk 4]
```diff
@@ -15,6 +16,12 @@ const routes = [
     name: 'create',
     path: '/create',
     component: Create
+  },
+
+  {
+    path: '/:catchAll(.*)',
+    component: PageNotFound,
+    name: 'NotFound'
```
Comment (ydcjeff) [inline src/main.js:24]:
```suggestion
    name: 'NotFound',
    path: '/:catchAll(.*)',
    component: PageNotFound
```

File: src/views/PageNotFound.vue [hunk 5]
```diff
@@ -0,0 +1,64 @@
+<template>
+  <main>
+    <NavBar />
+    <div class="heading">
+      <h1>404</h1>
+      <h3>Page Not Found</h3>
```
Comment (ydcjeff) [inline src/views/PageNotFound.vue:6]:
```suggestion
      <h1>404 Page Not Found</h1>
```

File: src/views/PageNotFound.vue [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+      <h3>Oops! We couldn't find that page.</h3>
```
Comment (ydcjeff) [inline src/views/PageNotFound.vue:7]:
```suggestion
      <p>Oops! We couldn't find that page.</p>
```

File: src/views/PageNotFound.vue [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+      <p>
+        Go back to the
+        <RouterLink to="/" class="link alt" :style="learnMore"
+          >homepage</RouterLink
+        >
```
Comment (ydcjeff) [inline src/views/PageNotFound.vue:12]:
```suggestion
        >.
```

File: src/views/PageNotFound.vue [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+      store.config.output_dir = './logs'
+      store.config.log_every_iters = 2
+    })
+  }
+}
+</script>
+
+<style scoped>
+h1 {
+  color: orange;
```
Comment (ydcjeff) [inline src/views/PageNotFound.vue:53]:
```suggestion
  color: var(--c-brand-red);
```

**Pull request merged**